### PR TITLE
Add logging and scope for ViewComponent

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        {
+            _viewComponentExecuting = LoggerMessage.Define<string>(
+                LogLevel.Verbose,
+                1,
+                "Executing view component {ViewComponentName}");
+
+            _viewComponentExecuted = LoggerMessage.Define<string, double, string>(
+                LogLevel.Verbose,
+                2,
+                "Executing view component {ViewComponentName} {ElapsedMilliseconds}ms elapsed returned " +
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs]: `Executed`? also `and returned`?

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            }
+
+            public IEnumerable<KeyValuePair<string, object>> GetValues()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override string ToString()
+            {
+                return base.ToString();
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs]: did you mean to return the view component name here?

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs]: sure

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs (hunk 2 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs]: yes :+1:

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Logging/DefaultViewComponentInvokerLoggerExtensions.cs]: Need to add the pairs too

**Comment by kichalla** [comment]
:shipit:

**Pull request closed without merging**